runtime/common/ObserveResult.h

```diff
@@ -77,8 +77,11 @@ class observe_result {
     // on more than 1 qubit can be <ZIII...III>
     auto numQubits = spinOp.num_qubits();
     auto termStr = term.to_string(false);
-    if (!data.has_expectation(termStr) && termStr.size() == 1 && numQubits > 1)
-      for (std::size_t i = 1; i < numQubits; i++)
+    // Expand the string representation of the term to match the number of
+    // qubits of the overall spin_op this result represents by appending
+    // identity ops.
+    if (!data.has_expectation(termStr) && termStr.size() < numQubits)
```

Comment:
Nit: I don't think the 2nd expression is required (because it is handled in your for loop below), so it could be simplified if you wanted:
```suggestion
    if (!data.has_expectation(termStr))
```

Reply:
Good point. Thank!